samcli/lib/init/template_modifiers/application_insights_template_modifier.py

```diff
@@ -0,0 +1,92 @@
+"""
+Class used to parse and update template when application-insights is enabled
+"""
+import logging
+from typing import Any
+import ruamel.yaml
+from ruamel.yaml import YAML
+from ruamel.yaml.comments import CommentedMap
+from samcli.lib.init.template_modifiers.cli_template_modifier import TemplateModifier
+
+LOG = logging.getLogger(__name__)
+yaml = YAML()
+# set ignore aliases to true
```

Comment:
Can we add more details why we are setting this config?

Reply:
Sure, will do. 
FYI, this is to avoid automatic usage of YAML aliases. In this case, yaml aliases is applied to self.RESOURCE_GROUP_NAME which is used twice. 
I tried to reference resourceGroupName from Resource group logical ID. However, I keep getting this error "Requested attribute Name must be a readonly property in schema for AWS::ResourceGroups::Group" 
Looks like we cannot use reference for Name attribute